ci/script.bash

```diff
@@ -12,3 +12,8 @@ if [[ $TARGET != arm-unknown-linux-gnueabihf ]] && [[ $TARGET != aarch64-unknown
     # Run 'bat' on its own source code and the README
     cargo run --target "$TARGET" -- src/bin/bat/main.rs README.md --paging=never
 fi
+
+# Check bat-as-a-library, which has a smaller set of dependencies
+cargo check --target "$TARGET" --verbose --lib --no-default-features
+cargo check --target "$TARGET" --verbose --lib --no-default-features --features git
+cargo check --target "$TARGET" --verbose --lib --no-default-features --features paging
```

Comment:
Does it make sense to also check `--no-default-features --features git,paging` explicitly, even if this is the combination which is also included in the default feature set?

Similarly, shouldn't the unit tests also be executed for all feature set combinations? Or is there a guarantee that they would also run if (1) `cargo check` for that feature set succeeds and (2) `cargo test` succeeds for all features enabled? I would assume that there is no such guarantee?

Reply:
> Does it make sense to also check `--no-default-features --features git,paging` explicitly, even if this is the combination which is also included in the default feature set?

In this case these two features are very orthogonal; no module contains both `cfg(feature = "git")` and `cfg(feature = "paging")` in the same module. But these library-only check builds are very fast once the dependencies have been built by previous builds. `--lib --no-default-features --features git,paging` only takes 0.7 seconds on my machine so we might as well &mdash; I've added it. But in general I haven't found exhaustively checking combinations of orthogonal features to be valuable in my library projects.

> Similarly, shouldn't the unit tests also be executed for all feature set combinations?

In https://github.com/dtolnay/syn we do `check` builds of various common feature combinations but only `test` with all features enabled, and that has been working well. Introducing more `cfg(...)` gunk in the test suite has tended not to be valuable.

Reply:
Thanks!